resteasy-core/src/main/java/org/jboss/resteasy/specimpl/ResteasyUriBuilderImpl.java

```diff
@@ -607,6 +604,8 @@ protected StringBuilder replaceParameter(Map<String, ? extends Object> paramMap,
             }
             Object value = paramMap.get(param);
             String stringValue = value != null ? value.toString() : null;
+            String regexValue = regexEval(string, stringValue, param);
+            stringValue = regexValue;
```

Comment:
Any reason not to just do:
```java
String stringValue = value != null ? regexValue(string, value.toString(), param) : null;
```

Reply:
an IllegalArgumentException should be thrown if value is null.  regexEval catches this.